File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap freqMapStringS = getCharacters(s);
        HashMap freqMapStringT = getCharacters(t);
        return freqMapStringS.equals(freqMapStringT);
    }
    
    public HashMap<Character, Long> getCharacters(String str) {
        HashMap<Character, Long> freqMap = new HashMap();
        for (char letter : str.toCharArray()){
            if(!freqMap.containsKey(letter)) {
                freqMap.put(letter, 1L);
            } else {
                freqMap.put(letter, freqMap.get(letter) + 1);
            } 
        }
        return freqMap;
    }    
}

File: 242-valid-anagram/242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        freqMapS = self.getFrequency(s)
        freqMapT = self.getFrequency(t)
        return freqMapS == freqMapT
        
    def getFrequency(self, word):
        freqMap = {}
        for char in word:
            if char in freqMap:
                freqMap[char] += 1
            else:
                freqMap[char] = 1
        return freqMap
    
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        